pkg/etcd/snapshot.go

```diff
@@ -540,34 +544,42 @@ func (e *ETCD) ListSnapshots(ctx context.Context) (map[string]snapshotFile, erro
 		if err != nil {
 			return nil, err
 		}
-		snapshotFiles = sfs
+		for k, sf := range sfs {
+			esf := k3s.NewETCDSnapshotFile("", k, k3s.ETCDSnapshotFile{})
+			sf.toETCDSnapshotFile(esf)
+			snapshotFiles.Items = append(snapshotFiles.Items, *esf)
+		}
 	}
 
 	sfs, err := e.listLocalSnapshots()
 	if err != nil {
 		return nil, err
 	}
 	for k, sf := range sfs {
-		snapshotFiles[k] = sf
+		esf := k3s.NewETCDSnapshotFile("", k, k3s.ETCDSnapshotFile{})
+		sf.toETCDSnapshotFile(esf)
+		snapshotFiles.Items = append(snapshotFiles.Items, *esf)
 	}
 
-	return snapshotFiles, err
+	return snapshotFiles, nil
 }
 
 // DeleteSnapshots removes the given snapshots from local storage and S3.
-func (e *ETCD) DeleteSnapshots(ctx context.Context, snapshots []string) error {
+func (e *ETCD) DeleteSnapshots(ctx context.Context, snapshots []string) ([]string, error) {
```

Comment:
Are the changes to this function about reporting back to client success now?

Reply:
Yeah, we have to send the list of created/deleted snapshots back to the client, since they don't get to see the log messages on the console.